app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -220,6 +222,17 @@ class LuciBuildService {
     );
   }
 
+  /// Get dimensions from target properties.
+  List<RequestedDimension> getDimensions(Map<String, dynamic>? properties) {
```

Comment:
This can be done later since P0, but can we move all the dimension logic be moved into model/ci_yaml/target.dart?